### ISSUE TITLE
Create a Pub/Sub Spring Integration sample that uses JSON payloads

### ISSUE THREAD

**Issue by meltsufin** [spring-cloud/spring-cloud-gcp#831]

**Comment by nbrinks** [comment]
I think this would be a great idea! I would also love to see a JSON sample without spring-integration as well. I am using 1.1.0.BUILD-SNAPSHOT and following the progress of this library closely. I started out with the Spring Integration sample but the plain `PubSubTemplate` fit my use case better. 

Here is some relevant feedback:

I am using Jackson in my project, which automatically enabled the `JacksonPubSubMessageConverter` thanks to `org.springframework.cloud.gcp.autoconfigure.pubsub.GcpPubSubAutoConfiguration`. This was a pleasant surprise. 👍

I think the most confusing part I ran into was that, AFAIK, I can publish messages using this method `org.springframework.cloud.gcp.pubsub.core.publisher.PubSubPublisherTemplate#publish(java.lang.String, T, java.util.Map<java.lang.String,java.lang.String>)` and the payload will be automatically converted thanks to `JacksonPubSubMessageConverter` but when I subscribe to messages via `org.springframework.cloud.gcp.pubsub.core.PubSubTemplate#subscribe` the received messages (via MessageReceiver interface) are not utilizing any message converter and I have to parse them manually. Perhaps this is not the intended way to subscribe to incoming messages. I think a sample would clear things up.

**Comment by dhoard** [comment]
I would guess in most people's use cases, the messages are in the same format, for example JSON objects. An auto-configured message converter makes sense for these scenarios.  

But, what about the scenario where messages in a topic may not be the same type? JSON vs XML? This use case is not common, not ideal, and quite frankly bad practice for a topic ... but it may occur where legacy systems are involved.

**Comment by dhoard** [comment]
Possible solution ...

1. Add the`PubSubMessageConverter` to the `PubSubSubscriberTemplate`

2. Create a Spring Pub / Sub specific `PubSubMessageReceiver` interface that has the `PubSubMessageConverter` as a parameter ...

```
public interface PubSubMessageReceiver {
	
	void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer ackReplyConsumer, PubSubMessageConverter pubSubMessageConverter);
}
```

3. Add a new `subscribe` method into the `PubSubSubscriberOperations` and `PubSubSubscriberTemplate` that accepts the new `PubSubMessageReceiver` ...

```
@Override
public Subscriber subscribe(String subscription, PubSubMessageReceiver pubSubMessageReceiver) {
    Assert.hasText(subscription, "The subscription can't be empty.");
    Assert.notNull(pubSubMessageReceiver, "The pubSubMessageReceiver can't be null.");
    
    return subscribe(subscription,
                    new MessageReceiver() {
                        @Override
                        public void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer ackReplyConsumer) {
                            pubSubMessageReceiver.receiveMessage(pubsubMessage, ackReplyConsumer, getMessasgeConverter());
                        }
                    });
}
```

**Comment by dzou** [comment]
Fixed by #1140
